### DISCUSSION TITLE
Proposal: Provide a video API to control the recording

### DISCUSSION THREAD

**Discussion by Lcfvs** [cypress-io/cypress#16706] [General]
Hi,

Is it possible to provide an API, callable during a run, to control when to start to record the videos, please?

The most common use is to be able to skip the cypress warmup phase and maybe some tweaks to hide the UI, when used to make some video demos.

**Comment by Lcfvs** [comment]
Since cypress starts to record from the beginning (then before the integration scripts), it can maybe need a new option, e.g. `"videoControlled": true` to distinct this specific behavior.

**Comment by jennifer-shehane** [comment]
You have access to the video after the spec file runs, so can use that to splice out any amount of the video timing. ['after:spec'](https://docs.cypress.io/api/plugins/after-spec-api#Examples)

There's an example of recording a hi res video without the Command Log in this repo also: https://github.com/bahmutov/cypress-movie and some other video experimental shown in this blog post: https://www.cypress.io/blog/2021/04/01/make-cypress-videos-look-old-school/

**Reply by Lcfvs** [reply]
I digged a few into that plugin code and it makes the things similarly than my solution. ;)

I understand the idea to cut it, after, I just suggest to have a way to simply control when to record, instead of to have the need to calculate the delay to cut & cut it afterwards.

It can be really simpler & comfortable. :P
